File: 2021/tests/exercises/test_exercise_8.py
from adventofcode.exercises import exercise_8
from adventofcode import utils

TEST_DATA = """be cfbegad cbdgef fgaecd cgeb fdcge agebfd fecdb fabcd edb | fdgacbe cefdb cefbgd gcbe
edbfga begcd cbg gc gcadebf fbgde acbgfd abcde gfcbed gfec | fcgedb cgb dgebacf gc
fgaebd cg bdaec gdafb agbcfd gdcbef bgcad gfac gcb cdgabef | cg cg fdcagb cbg
fbegcd cbd adcefb dageb afcb bc aefdc ecdab fgdeca fcdbega | efabcd cedba gadfec cb
aecbfdg fbg gf bafeg dbefa fcge gcbea fcaegb dgceab fcbdga | gecf egdcabf bgf bfgea
fgeab ca afcebg bdacfeg cfaedg gcfdb baec bfadeg bafgc acf | gebdcfa ecba ca fadegcb
dbcfg fgd bdegcaf fgec aegbdf ecdfab fbedc dacgb gdcebf gf | cefg dcbef fcge gbcadfe
bdfegc cbegaf gecbf dfcage bdacg ed bedf ced adcbefg gebcd | ed bcgafe cdgba cbgef
egadfb cdbfeg cegd fecab cgb gbdefca cg fgcdab egfdb bfceg | gbdfcae bgc cg cgb
gcafb gcf dcaebfg ecagb gf abcdeg gaef cafbge fdbac fegbdc | fgae cfgab fg bagce"""


def test_example_on_exercise_a():
    assert exercise_8.run_a(TEST_DATA) == 26


def test_returns_correct_answer_on_exercise_a():
    input_data = utils.load_data(exercise_8.EXERCISE)
    assert exercise_8.run_a(input_data) == 355


def test_example_on_exercise_b():
    assert exercise_8.run_b("acedgfb cdfbe gcdfa fbcad dab cefabd cdfgeb eafb cagedb ab | cdfeb fcadb cdfeb cdbaf") == 5353
    assert exercise_8.run_b("bdfegc cbegaf gecbf dfcage bdacg ed bedf ced adcbefg gebcd | ed bcgafe cdgba cbgef") == 1625
    assert exercise_8.run_b("fbegcd cbd adcefb dageb afcb bc aefdc ecdab fgdeca fcdbega | efabcd cedba gadfec cb") == 9361
    assert exercise_8.run_b(TEST_DATA) == 61229

def test_returns_correct_answer_on_exercise_b():
    input_data = utils.load_data(exercise_8.EXERCISE)
    assert exercise_8.run_b(input_data) == 983030

File: 2024/adventofcode/tests/test_exercise_8.rs
use adventofcode::exercises::exercise_8;
use adventofcode::exercises::utils;

const TEST_DATA: &str = "............
........0...
.....0......
.......0....
....0.......
......A.....
............
............
........A...
.........A..
............
............";

#[test]
fn test_a_returns_correct_answer_for_test_data() {
    assert_eq!(exercise_8::run_a(TEST_DATA), 14);
}

#[test]
fn test_a_returns_correct_answer_on_exercise_a() {
    let input_data = utils::load_data(exercise_8::EXERCISE);
    assert_eq!(exercise_8::run_a(&input_data), 278);
}

#[test]
fn test_b_returns_correct_answer_for_test_data() {
    assert_eq!(exercise_8::run_b(TEST_DATA), 34);
}

#[test]
fn test_b_returns_correct_answer_on_exercise_b() {
    let input_data = utils::load_data(exercise_8::EXERCISE);
    assert_eq!(exercise_8::run_b(&input_data), 1067);
}
